components/edit-collective/sections/MemberForm.js

```diff
@@ -119,55 +88,77 @@ const MemberForm = props => {
+                      defaultValue={getOptions([memberRole])[0]}
+                      onBlur={() => form.setFieldTouched(field.name, true)}
+                      onChange={({ value }) => {
+                        form.setFieldValue(field.name, value);
                         setMemberRole(value);
-                      }
-                    }}
+                      }}
+                      options={getOptions([roles.ADMIN, roles.MEMBER, roles.ACCOUNTANT])}
+                    />
+                    {hasRoleDescription(memberRole) && (
+                      <Flex mb={3}>
+                        <Box mx={1} mt={1} fontSize="12px" color="black.600" fontStyle="italic">
+                          <MemberRoleDescription role={memberRole} />
+                        </Box>
+                      </Flex>
+                    )}
+                  </React.Fragment>
+                )}
+              </StyledInputFormikField>
+              <StyledInputFormikField
+                name="description"
+                htmlFor="memberForm-description"
+                label={<P fontWeight="bold">{intl.formatMessage(memberFormMessages.descriptionLabel)}</P>}
+                mt={3}
+              >
+                {({ field }) => <StyledInput {...field} />}
+              </StyledInputFormikField>
+              <StyledInputFormikField
+                name="since"
+                htmlFor="memberForm-since"
+                inputType="date"
+                label={<P fontWeight="bold">{intl.formatMessage(memberFormMessages.sinceLabel)}</P>}
+                mt={3}
+              >
+                {({ form, field }) => (
+                  <StyledInput
+                    {...field}
+                    required
+                    value={dayjs(field.value).format('YYYY-MM-DD')}
                     onKeyDown={event => {
                       if (event.key === 'Backspace') {
                         event.preventDefault();
                       }
                     }}
-                    overflow="hidden"
-                    required={field.required}
+                    onChange={event => {
+                      const value = event.target.value;
+                      if (dayjs(value).isValid()) {
+                        form.setFieldValue(field.name, dayjs(value).toISOString());
+                      } else {
+                        form.setFieldValue(field.name, null);
+                      }
+                    }}
                   />
-                  {field.name === 'role' && hasRoleDescription(memberRole) && (
-                    <Flex mb={3}>
-                      <Box mx={1} mt={1} fontSize="12px" color="black.600" fontStyle="italic">
-                        <MemberRoleDescription role={memberRole} />
-                      </Box>
-                    </Flex>
-                  )}
-                </Container>
-              ))}
+                )}
+              </StyledInputFormikField>
```

Comment:
This date input is acting funny when typed in directly:

![Peek 2022-04-19 11-37](https://user-images.githubusercontent.com/1556356/163975480-7bc4774f-8d77-499d-ba4d-eda2111ce426.gif)

@SudharakaP could you help me to review this date input behavior and assist @snitin315 to get things right?

Reply:
@Betree : I've investigated this problem and the reason that this kinda weird behavior is the result of how the HTML5 `date` field is handled. By default (unless you set `max` and `min` attributes) the year can take upto 6 digits. That means from the HTML5 perspective something like 110111 is a valid date. However this isn't valid from the `dayjs` perspective (it needs to be 4 digits for year). What we are doing here is that we are formatting using `dayjs.format` method before setting the `value` attribute. Each time we type something it tries to do this `dayjs` formatting (of years such at 110111) and fails, defaults to `value` being `undefined` and hence the weird behavior.

Also there's a slight difference in how `date` field is handled in Chrome and Firefox. In Firefox if you set `min` and `max` attributes the date picker will only let you select within this range, but user can still type 6 digit years. In chrome setting the `min` and `max` prevents user from typing beyond the specified range as well. 

I suggest the following solution. When we are setting the `value` we need to make sure it's according to what HTML5 expects. Not what dayjs expects. However we should also set `min` and `max` to reasonable values. Finally we should probably do this validation within the `onChange` method at the end when the user submits the form; I don't see any point to do it on every single key press. 🤔 This will take care of this problem in both Chrome and Firefox. 

Sorry for the long winded explanation but to make things simpler I've added a commit to highlight what I am suggesting; https://github.com/opencollective/opencollective-frontend/pull/7697/commits/f2375f384011823e1d794dec1de7887e692358df

Let me know what you think.